422 generated when you put the wrong service port

#### Setup Details
 CIS Version : 2.10
 Build: f5networks/k8s-bigip-ctlr:latest       
 BIGIP Version: Big IP 15.1
 Agent Mode: AS3 
 Orchestration: K8S      
 Orchestration Version:       

#### Description  
When I create a virtualserver CRD with the "wrong" port, then I can see 422 from the logs of CIS and even if I delete the VS CRD, CIS never recovers (or it will take time to recover). I have to delete CIS to wrong again.   

#### Steps To Reproduce
1) Create a app working on port 8080 and the svc on port 80. 
```
apiVersion: apps/v1
kind: Deployment
metadata:
  name: app1
spec:
  replicas: 2
  selector:
    matchLabels:
      app: app1
  template:
    metadata:
      labels:
        app: app1
    spec:
      containers:
      - name: app1
        image: nginxdemos/nginx-hello:plain-text
        ports:
        - containerPort: 8080
---

apiVersion: v1
kind: Service
metadata:
  name: app1-svc
spec:
  ports:
  - port: 80
    targetPort: 8080
    protocol: TCP
    name: http
  selector:
    app: app1
```
2) Create the following VS pointing to port 80
```
apiVersion: "cis.f5.com/v1"
kind: VirtualServer
metadata:
  name: app1-vs
  labels:
    f5cr: "true"
spec:
  hostGroup: "test123"
  host: "app1.f5demo.test"
  ipamLabel: "prod"
  virtualServerName: "apps-vs"
  pools:
  - path: /
    service: app1-svc
    servicePort: 80
```

3) observe from the logs the 422 AS3 error. This is the ***FIRST PROBLEM*** as it shouldn't create a 422 for a `non-existent object app1_svc_80_default'

```
2022/09/13 06:52:54 [ERROR] [AS3] Raw response from Big-IP: map[code:422 declarationFullId: errors:[/cis-crd/Shared/apps_vs_80_app1_f5demo_test_policy/rules/0/actions/0/select/pool/use: contains path to non-existent object app1_svc_80_default] message:declaration is invalid] {"$schema":"https://raw.githubusercontent.com/F5Networks/f5-appsvcs-extension/master/schema/3.18.0/as3-schema-3.18.0-4.json","class":"AS3","declaration":{"cis-crd":{"Shared":{"app1_svc_8080_default":{"class":"Pool","members":[{"addressDiscovery":"static","serverAddresses":["10.244.140.84"],"servicePort":8080,"shareNodes":true},{"addressDiscovery":"static","serverAddresses":["10.244.196.137"],"servicePort":8080,"shareNodes":true}]},"apps_vs_80":{"source":"0.0.0.0/0","translateServerAddress":true,"translateServerPort":true,"class":"Service_HTTP","virtualAddresses":["10.1.10.170"],"virtualPort":80,"snat":"auto","policyEndpoint":"/cis-crd/Shared/apps_vs_80_app1_f5demo_test_policy"},"apps_vs_80_app1_f5demo_test_policy":{"class":"Endpoint_Policy","rules":[{"name":"vs_app1_f5demo_test_app1_svc_80_default","conditions":[{"type":"httpHeader","name":"host","event":"request","all":{"values":["app1.f5demo.test"],"operand":"equals"}}],"actions":[{"type":"forward","event":"request","select":{"pool":{"use":"app1_svc_80_default"}}}]}],"strategy":"first-match"},"class":"Application","template":"shared"},"class":"Tenant","defaultRouteDomain":0},"class":"ADC","controls":{"class":"Controls","userAgent":"CIS/v2.9.0 K8S/v1.22.7"},"id":"urn:uuid:B97DFADF-9F0D-4F6C-8D66-E9B52E593694","label":"CIS Declaration","remark":"Auto-generated by CIS","schemaVersion":"3.18.0"}}
2022/09/13 06:52:54 [ERROR] [AS3] Big-IP Responded with code: 422
```

4. Delete the VS and see that CIS never deletes the configuration and keeps on sending the same declaration to BIGIP, always getting back 422. Check on the last line of the logs that CIS mentions that it cannot find the VS you just deleted `2022/09/13 06:53:26 [DEBUG] VirtualServer Not Found: default/app1-vs` 

```
2022/09/13 06:53:24 [DEBUG] [AS3] posting request to https://10.1.1.5/mgmt/shared/appsvcs/declare/cis-crd
2022/09/13 06:53:26 [ERROR] [AS3] Raw response from Big-IP: map[code:422 declarationFullId: errors:[/cis-crd/Shared/apps_vs_80_app1_f5demo_test_policy/rules/0/actions/0/select/pool/use: contains path to non-existent object app1_svc_80_default] message:declaration is invalid] {"$schema":"https://raw.githubusercontent.com/F5Networks/f5-appsvcs-extension/master/schema/3.18.0/as3-schema-3.18.0-4.json","class":"AS3","declaration":{"cis-crd":{"Shared":{"app1_svc_8080_default":{"class":"Pool","members":[{"addressDiscovery":"static","serverAddresses":["10.244.140.84"],"servicePort":8080,"shareNodes":true},{"addressDiscovery":"static","serverAddresses":["10.244.196.137"],"servicePort":8080,"shareNodes":true}]},"apps_vs_80":{"source":"0.0.0.0/0","translateServerAddress":true,"translateServerPort":true,"class":"Service_HTTP","virtualAddresses":["10.1.10.170"],"virtualPort":80,"snat":"auto","policyEndpoint":"/cis-crd/Shared/apps_vs_80_app1_f5demo_test_policy"},"apps_vs_80_app1_f5demo_test_policy":{"class":"Endpoint_Policy","rules":[{"name":"vs_app1_f5demo_test_app1_svc_80_default","conditions":[{"type":"httpHeader","name":"host","event":"request","all":{"values":["app1.f5demo.test"],"operand":"equals"}}],"actions":[{"type":"forward","event":"request","select":{"pool":{"use":"app1_svc_80_default"}}}]}],"strategy":"first-match"},"class":"Application","template":"shared"},"class":"Tenant","defaultRouteDomain":0},"class":"ADC","controls":{"class":"Controls","userAgent":"CIS/v2.9.0 K8S/v1.22.7"},"id":"urn:uuid:B97DFADF-9F0D-4F6C-8D66-E9B52E593694","label":"CIS Declaration","remark":"Auto-generated by CIS","schemaVersion":"3.18.0"}}
2022/09/13 06:53:26 [ERROR] [AS3] Big-IP Responded with code: 422
2022/09/13 06:53:26 [DEBUG] [AS3] Posting failed tenants configuration in 30s seconds
2022/09/13 06:53:26 [DEBUG] VirtualServer Not Found: default/app1-vs 
```

#### Expected Result 
The declaration shouldn't be sent on BIGIP, if the Port number is wrong and VS CRD status should be updated with the error message. 
Having said that Port 80 is the correct portnumber for the service.

--- a/pkg/controller/resourceConfig.go
+++ b/pkg/controller/resourceConfig.go
@@ -426,10 +426,10 @@ func (ctlr *Controller) prepareRSConfigFromVirtualServer(
 			continue
 		}
 		framedPools[poolName] = struct{}{}
-		targetPort := intstr.IntOrString{IntVal: pl.ServicePort}
+		targetPort := ctlr.fetchTargetPort(vs.Namespace, pl.Service, pl.ServicePort)
 
 		if (intstr.IntOrString{}) == targetPort {
-			targetPort = ctlr.fetchTargetPort(vs.Namespace, pl.Service, pl.ServicePort)
+			targetPort = intstr.IntOrString{IntVal: pl.ServicePort}
 		}
 		svcNamespace := vs.Namespace
 		if pl.ServiceNamespace != "" {

--- a/pkg/controller/worker.go
+++ b/pkg/controller/worker.go
@@ -1518,6 +1518,10 @@ func (ctlr *Controller) updatePoolMembersForNodePort(
 					ctlr.getEndpointsForNodePort(svcPort.NodePort, pool.NodeMemberLabel)
 			}
 		}
+		//check if endpoints are found
+		if rsCfg.Pools[index].Members == nil {
+			log.Errorf("[CORE]Endpoints could not be fetched for service %v with targetPort %v", svcName, pool.ServicePort.IntVal)
+		}
 	}
 }
 
@@ -1544,6 +1548,10 @@ func (ctlr *Controller) updatePoolMembersForCluster(
 			}
 			rsCfg.MetaData.Active = true
 			rsCfg.Pools[index].Members = mems
+		}
+		//check if endpoints are found
+		if rsCfg.Pools[index].Members == nil {
+			log.Errorf("[CORE]Endpoints could not be fetched for service %v with targetPort %v", svcName, pool.ServicePort.IntVal)
 		}
 	}
 }

